addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -621,11 +624,17 @@ const wSnippetMenu = weSnippetEditor.SnippetsMenu.extend({
      * @param {Event} ev
      */
     _onTextHighlightClick(ev) {
+        // To be able to open the highlights grid immediately, we need to
+        // prevent the `_onClick()` handler from closing the widget (using
+        // the `_closeWidgets()` method) right after opening it.
+        ev.stopPropagation();
+        this._closeWidgets();
         const target = ev.currentTarget;
         this._handleTextOptions(target, [
             this._getOptionTextClass(target),
             "o_text_highlight_freehand_1",
-        ]);
+        ],
+        () => this.options.wysiwyg.odooEditor.dispatchEvent(new Event("show_text_highlights")));
```

Comment:
- Strange alignment IMO
- Maybe you could use the system around `_onOptionUpdate` instead of disaptching event through the odooEditor which is really not related to snippet options?

Reply:
> Maybe you could use the system around _onOptionUpdate instead of disaptching event through the odooEditor which is really not related to snippet options?

Yes, we can trigger an `"option_update"` from the editor linked to the activated text snippet.
Actually, I had to do it that way because it's only on `_handleTextOptions()` where I'm able to check if the highlight button was clicked for the first time in the toolbar, and also the system using the `odooEditor` event to link it with the options was already implemented in the same way to [activate link tools](https://github.com/odoo/odoo/blob/17.0/addons/web_editor/static/src/js/editor/snippets.options.js#L5901).

I updated the code now to use the `"option_update"` & `notify()` systems, thanks :+1: